render/respond.go

```diff
@@ -0,0 +1,111 @@
+package render
+
+import (
+	"bytes"
+	"encoding/json"
+	"encoding/xml"
+	"net/http"
+	"reflect"
+
+	"golang.org/x/net/context"
+)
+
+var Respond = DefaultRespond
+
+func DefaultRespond(ctx context.Context, w http.ResponseWriter, v interface{}) {
```

Comment:
If you pass a pointer it's going to be 4 bytes - non issue.

Middlewares being unable to add headers after blocks us from changing the headers in middlewares based on response content (like for example generic pager middleware - which would be super neat :D) 

Another limitation: right now middlewares are forced to write headers to the response, so a middleware can't override a header set by a previous middleware (or strip the header).